control/controlclient/auto.go

```diff
@@ -171,27 +173,45 @@ func (c *Auto) sendNewMapRequest() {
 	// If we're not already streaming a netmap, or if we're already stuck
 	// in a lite update, then tear down everything and start a new stream
 	// (which starts by sending a new map request)
-	if !c.inPollNetMap || c.inLiteMapUpdate || !c.loggedIn {
+	if !c.inPollNetMap || !c.loggedIn {
 		c.mu.Unlock()
 		c.cancelMapSafely()
 		return
 	}
+	if c.inLiteMapUpdate {
+		if c.liteMapUpdateCancels > 10 {
+			// We've canceled 10 of these, and thus aren't making
+			// progress; fall back to tearing everything down.
+			c.mu.Unlock()
+			c.cancelMapSafely()
```

Comment:
You probably want to clean up after yourself here like cancel the inprogress lite map update and reset the counter to 0